packages/shared/src/components/auth/PostAuthBanner.tsx

```diff
@@ -1,11 +1,116 @@
+import { useQuery } from '@tanstack/react-query';
+import classNames from 'classnames';
 import { feature } from '../../lib/featureManagement';
 import { checkIsBrowser, checkIsExtension, UserAgent } from '../../lib/func';
 import { AuthExtensionBanner } from './AuthExtensionBanner';
 import { AuthenticationBanner } from './AuthenticationBanner';
+import { getBasicUserInfo } from '../../graphql/users';
+import { OnboardingHeadline } from './OnboardingHeadline';
+import { sizeClasses } from '../ProfilePicture';
+import { Image, ImageType } from '../image/Image';
 import { useConditionalFeature } from '../../hooks';
+import { IconSize } from '../Icon';
+import {
+  getSocialReferrer,
+  socialCTA,
+  socialGradient,
+  socialIcon,
+  type SocialIconType,
+} from '../../lib/socialMedia';
+import { capitalize } from '../../lib/strings';
+import { useAuthContext } from '../../contexts/AuthContext';
+
+const geoToEmoji = (geo: string): string => {
+  return geo
+    .toUpperCase()
+    .split('')
+    .map((char) => String.fromCodePoint(char.charCodeAt(0) + 0x1f1a5))
+    .join('');
+};
+
+const GeoPersonalizedBanner = ({ geo }: { geo: string }): ReactElement => {
+  const emoji = geoToEmoji(geo);
+  const displayNames = new Intl.DisplayNames(['en'], { type: 'region' });
+  const country = displayNames.of(geo.toUpperCase());
+
+  return (
+    <AuthenticationBanner>
+      <span className="text-[3.5rem] leading-none">{emoji}</span>
+      <OnboardingHeadline
+        className={{
+          title: `typo-mega3`,
+          description: 'mb-8 typo-title3',
+        }}
+        title={`daily.dev is the fastest growing developer platform in ${country}!`}
+        description="We know how hard it is to be a developer. It doesn't have to be. Personalized news feed, dev community and search, much better than what's out there. Maybe ;)"
+      />
+    </AuthenticationBanner>
+  );
+};
+
+const SocialPersonalizedBanner = ({
+  site,
+}: {
+  site: SocialIconType;
+}): ReactElement => {
+  const Icon = socialIcon[site];
+  const gradient = socialGradient[site];
+  return (
+    <AuthenticationBanner>
+      <Icon size={IconSize.XXLarge} secondary={site === 'reddit'} />
+      <OnboardingHeadline
+        className={{
+          title: classNames('typo-mega3', gradient),
+          description: 'mb-8 typo-title3',
+        }}
+        pretitle={`Coming from ${capitalize(site)}?`}
+        {...socialCTA[site]}
+      />
+    </AuthenticationBanner>
+  );
+};
+
+const UserPersonalizedBanner = ({
+  userId,
+}: {
+  userId: string;
+}): ReactElement => {
+  const { data: user, isError } = useQuery({
+    queryKey: ['user_personalized_banner', userId],
+    queryFn: () => getBasicUserInfo(userId),
+  });
+
+  if (isError) {
+    return <AuthenticationBanner />;
+  }
+
+  const name = user?.name ? user?.name.split(' ')[0] : user?.username;
+
+  return (
+    <AuthenticationBanner>
+      {user?.image && (
+        <Image
```

Comment:
Should we use ProfileImage here? (or whatever it's called, it handles all classes already)